pkg/util/release/environment_test.go

```diff
@@ -0,0 +1,99 @@
+package release
+
+import (
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+	"testing"
+)
+
+func TestReleaseIsProgressing(t *testing.T) {
+	var tests = []struct {
+		title        string
+		achievedStep *shipper.AchievedStep
+		targetStep   int32
+		expected     bool
+	}{
+		{
+			"nil achieved step",
+			nil,
+			0,
+			true,
+		},
+		{
+			"empty achieved step",
+			&shipper.AchievedStep{},
+			0,
+			true,
+		},
+		{
+			"progressing release",
+			&shipper.AchievedStep{
+				Step: 0,
+				Name: "0",
+			},
+			2,
+			true,
+		},
+		{
+			"not progressing release",
+			&shipper.AchievedStep{
+				Step: 1,
+				Name: "1",
+			},
+			0,
+			false,
+		},
+	}
+	for _, test := range tests {
+		actual := ReleaseIsProgressing(test.achievedStep, test.targetStep)
+		if actual != test.expected {
+			t.Fatalf("testing %s: expected %t, actual %t", test.title, test.expected, actual)
+		}
+	}
+}
+
+func TestStepsFromAchievedStep(t *testing.T) {
+	var tests = []struct {
+		title        string
+		achievedStep *shipper.AchievedStep
+		targetStep   int32
+		expected     int
+	}{
+		{
+			"nil achieved step",
+			nil,
+			0,
+			0,
+		},
+		{
+			"empty achieved step",
+			&shipper.AchievedStep{},
+			0,
+			0,
+		},
+		{
+			"achieved step",
+			&shipper.AchievedStep{
+				Step: 0,
+				Name: "0",
```

Comment:
This is very nitpicky, but I'd have omitted the Name in the struct, since it's not used. Using it kinda denotes the intention that it *would* be used, but it's not the case here, so I personally find it distracting.